framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,143 @@
+                        "used when \"furthest_from_zero\" is set to "
+                        "\"true\"");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero")),
+    _preserve_sign(_furthest_from_zero && getParam<bool>("preserve_sign"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+
+  if (getParam<bool>("preserve_sign") != _preserve_sign)
+    mooseDoOnce(
+        mooseWarning("================================ ",
+                     name(),
+                     " ================================\n",
+                     "  block = VectorPostprocessors\n",
+                     "  type = ElementVariablesDifferencesMax\n",
+                     "\n",
+                     "\"preserve_sign\" was set to \"true\" but \"furthest_from_zero\" is not.\n",
+                     "This option will be ignored.\n",
+                     "================================ ",
+                     name(),
+                     " ================================"));
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference and its sign
+    Real difference = _a[qp] - _b[qp];
+    bool is_negative = (difference < 0);
+
+    // Dirty little absolute function
+    if (_furthest_from_zero)
+      difference = is_negative ? -difference : difference;
+
+    // Assign the appropriate values if a new maximum is found
+    if (difference > _max_difference[0])
+    {
+      _max_difference[0] = difference;
+      _max_difference_is_negative = is_negative;
+      _a_value[0] = _a[qp];
+      _b_value[0] = _b[qp];
+
+      _position_x[0] = _q_point[qp](0);
+      _position_y[0] = _q_point[qp](1);
+      _position_z[0] = _q_point[qp](2);
+    }
+  }
+}
+
+void
+ElementVariablesDifferenceMax::finalize()
+{
+  // Gather all the parameters
+  gatherProxyValueMax(_max_difference[0], _a_value[0]);
+  gatherProxyValueMax(_max_difference[0], _b_value[0]);
+  gatherProxyValueMax(_max_difference[0], _max_difference[0]);
+  gatherProxyValueMax(_max_difference[0], _max_difference_is_negative);
+  gatherProxyValueMax(_max_difference[0], _position_x[0]);
+  gatherProxyValueMax(_max_difference[0], _position_y[0]);
+  gatherProxyValueMax(_max_difference[0], _position_z[0]);
```

Comment:
Collect the relevant information associated with the largest difference: location and actual value in each field. I want to know more than just the largest difference, hence why this is a VectorPostprocessor instead of a regular Postprocessor.

None of the MPI sync commands work with libMesh::Point, so instead of writing an explicit version I split the coordinates into separate points.

Please let me know if there is a better method to accomplish this task!